security/pkg/nodeagent/sds/server.go

```diff
@@ -185,11 +189,17 @@ func (s *Server) initGatewaySdsService(options *Options) error {
 	}
 
 	go func() {
-		if err = s.grpcGatewayServer.Serve(s.grpcGatewayListener); err != nil {
-			log.Errorf("SDS grpc server for ingress gateway proxy failed to start: %v", err)
+		for {
+			// Retry if Serve() fails
+			log.Info("Start SDS grpc server for ingress gateway proxy")
+			if err = s.grpcGatewayServer.Serve(s.grpcGatewayListener); err != nil {
+				log.Errorf("SDS grpc server for ingress gateway proxy failed to start: %v", err)
+			}
+			s.grpcGatewayListener, err = setUpUds(options.IngressGatewayUDSPath)
+			if err != nil {
+				log.Errorf("SDS grpc server for ingress gateway proxy failed to set up UDS: %v", err)
```

Comment:
Same here

Reply:
"ingress gateway proxy failed to set up UDS" is different from " ingress gateway proxy failed to start".